### DISCUSSION TITLE
domainname not set

### DISCUSSION THREAD

**Discussion by dkirrane** [vmware/packer-examples-for-vsphere#298] [Help]
I have successfully build a `ubuntu-base-22.04.1-LTS` VM template with Packer and cloud-init as in the example.

However, when I then use the [Terraform example](https://github.com/vmware-samples/packer-examples-for-vsphere/blob/main/terraform/vsphere-virtual-machine/template-linux-cloud-init/metadata.yml) to create a new VM from it the `domainname` is not getting set.

I have tried update `metadata.yml` in a number of different settings like below but nothing seems to work:

#### metadata.yml
```
instance-id: vm_name
hostname: vm_name.domain
local-hostname: vm_name.domain
local_hostname: vm_name.domain
fqdn: vm_name.domain
```

On the vm I'd expect to see the domain name being: 
- `hostname --fqdn`
- `domainname`
- `hostnamectl`

**Comment by tenthirtyam** [comment]
Try:

```yaml
instance-id:
hostname:
prefer_fqdn_over_hostname: true
fqdn:
local-hostname:
```

**Reply by dkirrane** [reply]
@tenthirtyam ya just tried that same issue
```
$ hostname
foo
$ domainname
(none)
```

Also tried switch from cloud-init to terraform clone > customize option but that doesn't work at all. VM doesn't get an IP, it just seems to hang in the vSphere client remote console.
```
  clone {
    template_uuid = data.vsphere_virtual_machine.template.id
    timeout       = 30
    customize {
      timeout = 10
      linux_options {
        host_name = var.vm_name
        domain    = var.domain
      }
        network_interface {
          ipv4_address = var.ip_address
          ipv4_netmask = var.ip_netmask
        }
        ipv4_gateway    = var.gateway
        dns_server_list = var.dns_server_list
        dns_suffix_list = var.dns_suffix_list
    }
  }
```

**Reply by tenthirtyam** [reply]
I won't have a chance to look into it until mid to late November.
